src/services/machines/onboardingMachineService.ts

```diff
@@ -121,20 +126,77 @@ export const storePIDCredentials = async (context: Pick<OnboardingMachineContext
 };
 
 export const setupWallet = async (
-  context: Pick<OnboardingMachineContext, 'pinCode' | 'emailAddress' | 'name' | 'biometricsEnabled' | 'pidCredentials' | 'countryCode'>,
+    context: Pick<OnboardingMachineContext, 'pinCode' | 'emailAddress' | 'name' | 'biometricsEnabled' | 'pidCredentials' | 'countryCode'| 'credentialData'>
 ): Promise<WalletSetupServiceResult> => {
   const {pinCode} = context;
   const setup = await Promise.all([
     storagePersistPin({
       value: pinCode,
     }),
+    createSelfIssuedCredential(context)
+        .then((credential) =>
+           agent.ibAddCredentialBranding({
+            vcHash: credential.hash,
+            issuerCorrelationId: credential.issuerCorrelationId,
+            localeBranding: [SphereonWalletIdentityBranding]
+          })
+        ),
     storeUser(context),
     // Make sure we never finish before the timeout, to ensure the UI doesn't navigate too fast for a user between screens
     new Promise(resolve => setTimeout(() => resolve(true), 1000)),
   ]);
 
-  await store.dispatch<any>(login(setup[1].storedUser.id));
-  return setup[1];
+  await store.dispatch<any>(login(setup[2].storedUser.id));
+  return setup[2];
+};
+
+const createSelfIssuedCredential = async (
+    context: Pick<OnboardingMachineContext, 'emailAddress' | 'name' | 'credentialData'>
+): Promise<DigitalCredential> => {
+  const {emailAddress, name, credentialData} = context;
+  const identifier: IIdentifier = await getOrCreatePrimaryIdentifier(
+      {
+        method: credentialData.didMethod,
+        createOpts: {options: credentialData.didOptions},
+      },
+      agentContext,
+  );
+
+  const names = parseFullName(name);
```

Comment:
Can this be undefined? Or fields within names.
(I cannot see in parseFullName here)

Reply:
no default is an empty string in the onboarding machine